Dockerfile

```
@@ -22,6 +22,7 @@ RUN apt-get update \
     && ssh-keyscan -H github.com > /root/ssh/known_hosts
 
 ENV PYTHON=python3
+RUN npm install -g node-gyp
 COPY package.json yarn.lock /usr/src/app/
 RUN npm install typescript@4.9.5 -g
```

best to install all "global" packages in a single step,
before package.json copy (to avoid invalidatoin),
and specifying the node-gyp version (just like we do for typescript)

```suggestion
RUN npm install -g \
    node-gyp@11.0.0 \
    typescript@4.9.5
COPY package.json yarn.lock /usr/src/app/
```

---

ping